tools/config-generator/main_test.go

```diff
@@ -43,3 +54,869 @@ func TestOutputConfig(t *testing.T) {
+		t.Errorf("Unexpected env value: (-got +want)\n%s", diff)
+	}
+
+	job = baseProwJobTemplateData{}
+	job.addEnvToJob("num", "42")
+	if diff := cmp.Diff(job.Env[0], "- name: num"); diff != "" {
+		t.Errorf("Unexpected env name: (-got +want)\n%s", diff)
+	}
+	if diff := cmp.Diff(job.Env[1], "  value: \"42\""); diff != "" {
+		t.Errorf("Unexpected env value: (-got +want)\n%s", diff)
+	}
+}
+
+func TestAddLabelToJob(t *testing.T) {
+	SetupForTesting()
+	job := baseProwJobTemplateData{}
+	addLabelToJob(&job, "foo", "bar")
+	labelString := fmt.Sprintf("%v", job.Labels)
+	expected := "[foo: bar]"
+	if diff := cmp.Diff(labelString, expected); diff != "" {
+		t.Errorf("Unexpected label string: (-got +want)\n%s", diff)
+	}
+}
+
+func TestAddMonitoringPubsubLabelsToJob(t *testing.T) {
+	SetupForTesting()
+	job := baseProwJobTemplateData{}
+	addMonitoringPubsubLabelsToJob(&job, "foobar")
+	expected := []string{
+		"prow.k8s.io/pubsub.project: knative-tests",
+		"prow.k8s.io/pubsub.topic: knative-monitoring",
+		"prow.k8s.io/pubsub.runID: foobar",
+	}
+	for i := range expected {
+		if diff := cmp.Diff(job.Labels[i], expected[i]); diff != "" {
+			t.Errorf("Unexpected pubsub label: (-got +want)\n%s", diff)
+		}
+	}
+}
+
+func TestAddVolumeToJob(t *testing.T) {
+	SetupForTesting()
+	mountPath := "somePath"
+	name := "foo"
+	content := []string{"bar", "baz"}
+
+	job := baseProwJobTemplateData{}
+	isSecret := false
+	addVolumeToJob(&job, mountPath, name, isSecret, content)
+	expectedVolumeMounts := []string{
+		"- name: foo",
+		"  mountPath: somePath",
+	}
+	for i := range expectedVolumeMounts {
+		if diff := cmp.Diff(job.VolumeMounts[i], expectedVolumeMounts[i]); diff != "" {
+			t.Errorf("Unexpected volume mount: (-got +want)\n%s", diff)
+		}
+	}
+	expectedVolumes := []string{
+		"- name: foo",
+		"  bar",
+		"  baz",
+	}
+	for i := range expectedVolumes {
+		if diff := cmp.Diff(job.Volumes[i], expectedVolumes[i]); diff != "" {
+			t.Errorf("Unexpected volume: (-got +want)\n%s", diff)
+		}
+	}
+
+	job = baseProwJobTemplateData{}
+	isSecret = true
+	addVolumeToJob(&job, mountPath, name, isSecret, content)
+	expectedVolumeMounts = []string{
+		"- name: foo",
+		"  mountPath: somePath",
+		"  readOnly: true",
+	}
+	for i := range expectedVolumeMounts {
+		if diff := cmp.Diff(job.VolumeMounts[i], expectedVolumeMounts[i]); diff != "" {
+			t.Errorf("Unexpected volume mount: (-got +want)\n%s", diff)
+		}
+	}
+	expectedVolumes = []string{
+		"- name: foo",
+		"  secret:",
+		"    secretName: foo",
+		"  bar",
+		"  baz",
+	}
+	for i := range expectedVolumes {
+		if diff := cmp.Diff(job.Volumes[i], expectedVolumes[i]); diff != "" {
+			t.Errorf("Unexpected volume: (-got +want)\n%s", diff)
+		}
+	}
+}
+
+func TestConfigureServiceAccountForJob(t *testing.T) {
+	SetupForTesting()
+	job := baseProwJobTemplateData{ServiceAccount: ""}
+	configureServiceAccountForJob(&job)
+	if logFatalCalls != 0 || len(job.Volumes) != 0 {
```

Comment:
I'm a little concerned about this global variable used for testing, if all these tests are running in parallel, would this still be valid?

Reply:
Great observation. I would definitely say these tests are vulnerable to race conditions if run in parallel. It's not just this global variable, it's all of them. That might be another good argument for rewriting the entire tool.